Update the linker script to support RW/ZI/Heap span across RAM banks

deepikabhavnani opened the pull request:
### Description
As per EWARM Development guide 

A region consists of one or several memory ranges, where each memory range
consists of a continuous sequence of bytes in a specific memory. Several ranges
can be combined by using region expressions. Note that those ranges do
not need to be consecutive or even in the same memory.

RAM region here is made as combination of 2-RAM banks
define region RAM_region = mem:[from __SRAM2_start__ to __SRAM2_end__]
                          | mem:[from __SRAM1_start__ to __SRAM1_end__];

block can be placed in specific region which can have several ranges.
place in RAM_region                    {  block RW };
place in RAM_region                    {  block ZI };
place in RAM_region                    {  block HEAP };

### Pull request type
    [X] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

jeromecoutant reviewed (commented):
I could approve, but who takes the action to align all icf files for all targets ?

Commit:
Update the linker script to support RW/ZI/Heap span across RAM banks

As per EWARM Development guide
A region consists of one or several memory ranges, where each memory range
consists of a continuous sequence of bytes in a specific memory. Several ranges
can be combined by using region expressions. Note that those ranges do
not need to be consecutive or even in the same memory.

RAM region here is made as combination of 2-RAM banks
define region RAM_region = mem:[from __SRAM2_start__ to __SRAM2_end__]
                          | mem:[from __SRAM1_start__ to __SRAM1_end__];

block can be placed in specific region which can have several ranges.
place in RAM_region                    {  block RW };
place in RAM_region                    {  block ZI };
place in RAM_region                    {  block HEAP };

Commit:
Make heap to be of expanding size
